### ISSUE TITLE
[operating_systems] Document where to find Red Hat products' source code

### ISSUE THREAD

**Issue by ekultails** [ekultails/rootpages#187]
All of Red Hat's products are open source. The downstream source code for the subscription packages can be found here: http://ftp.redhat.com/redhat/

RHEL source code in particular can be found here: http://ftp.redhat.com/redhat/linux/enterprise/

**Comment by m33k** [comment]
Thanks for sharing! Do you know if the commandline tools are built from source too or is that handled by the upstream developers? I was checking in http://ftp.redhat.com/redhat/linux/enterprise/7Server/en/ for clitools, but it seems to be src.rpm

**Comment by ekultails** [comment]
@m33k Yes, almost everything from Red Hat is provided as source RPMs. Those are cpio archives and can be extracted using the `rpm2cpio` utility.

What package are you looking for exactly? Starting with RHEL 7, all RHEL source code now lives under the CentOS project as git repositories. You can find that [here](https://git.centos.org/). For example, the RPM spec file for Bash can be found here [1]. It downloads the source code from the upstream project. Then it applies Enterprise Linux specific patches on top of that [2].

1. https://git.centos.org/rpms/bash/blob/c8/f/SPECS/bash.spec#_13
2. https://git.centos.org/rpms/bash/blob/c8/f/SOURCES

**Comment by ekultails** [comment]
I updated my first post to clarify that the essential BaseOS source code is hosted elsewhere.

**Comment by m33k** [comment]
@ekultails , sorry for the delay. I was interested in the cli tools for the most part. Example, like things coming from procps or coreutils.

Both of which I was able to find the repo via the links you mentioned, but there aren't any commits at this time.

https://git.centos.org/rpms/coreutils
https://git.centos.org/rpms/procps-ng

Thanks for the help.

**Comment by ekultails** [comment]
@m33k Checkout the branch that relates to the version you care about. That is where the source is for the RPM. Normally the actually source code is listed in the RPM spec file as `Source0` where it is downloaded upstream and then patches are applied ontop of that.

https://git.centos.org/rpms/coreutils/branches
https://git.centos.org/rpms/coreutils/blob/c8/f/SPECS/coreutils.spec#_8

**Comment by m33k** [comment]
Thanks!
